Userland/Games/GameOfLife/BoardWidget.cpp

```diff
@@ -178,3 +231,156 @@ Optional<Board::RowAndColumn> BoardWidget::get_row_and_column_for_point(int x, i
+
+    Vector<String> infinite_2 = {
+        "......O.",
+        "....O.OO",
+        "....O.O.",
+        "....O...",
+        "..O.....",
+        "O.O....."
+    };
+
+    Vector<String> infinite_3 = {
+        "OOO.O",
+        "O....",
+        "...OO",
+        ".OO.O",
+        "O.O.O"
+    };
+
+    Vector<String> simkin_glider_gun = {
+        "OO.....OO........................",
+        "OO.....OO........................",
+        ".................................",
+        "....OO...........................",
+        "....OO...........................",
+        ".................................",
+        ".................................",
+        ".................................",
+        ".................................",
+        "......................OO.OO......",
+        ".....................O.....O.....",
+        ".....................O......O..OO",
+        ".....................OOO...O...OO",
+        "..........................O......",
+        ".................................",
+        ".................................",
+        ".................................",
+        "....................OO...........",
+        "....................O............",
+        ".....................OOO.........",
+        ".......................O........."
+    };
+    Vector<String> gosper_glider_gun = {
+        "........................O...........",
+        "......................O.O...........",
+        "............OO......OO............OO",
+        "...........O...O....OO............OO",
+        "OO........O.....O...OO..............",
+        "OO........O...O.OO....O.O...........",
+        "..........O.....O.......O...........",
+        "...........O...O....................",
+        "............OO......................"
+    };
+
+    Vector<String> r_pentomino = {
+        ".OO",
+        "OO.",
+        ".O."
+    };
+
+    Vector<String> diehard = {
+        "......O.",
+        "OO......",
+        ".O...OOO"
+    };
+
+    Vector<String> acorn = {
+        ".O.....",
+        "...O...",
+        "OO..OOO"
+    };
+
+    add_pattern("Blinker", make<Pattern>(blinker));
+    add_pattern("Toad", make<Pattern>(toad));
+    add_pattern("Glider", make<Pattern>(glider));
+    add_pattern("Lightweight Spaceship", make<Pattern>(lightweight_spaceship));
+    add_pattern("Middleweight Spaceship", make<Pattern>(middleweight_spaceship));
+    add_pattern("Heavyweight Spaceship", make<Pattern>(heavyweight_spaceship));
+    add_pattern("Infinite 1", make<Pattern>(infinite_1));
+    add_pattern("Infinite 2", make<Pattern>(infinite_2));
+    add_pattern("Infinite 3", make<Pattern>(infinite_3));
+    add_pattern("R-Pentomino", make<Pattern>(r_pentomino));
+    add_pattern("Diehard", make<Pattern>(diehard));
+    add_pattern("Acorn", make<Pattern>(acorn));
+    add_pattern("Simkin's Glider Gun", make<Pattern>(simkin_glider_gun));
+    add_pattern("Gosper's Glider Gun", make<Pattern>(gosper_glider_gun));
```

Comment:
We `move()` each vector in the Pattern constructor, but I think we still need to move them here as well to avoid a copy?

Reply:
> We `move()` each vector in the Pattern constructor, but I think we still need to move them here as well to avoid a copy?

```     
    auto add_pattern = [&](String name, NonnullOwnPtr<Pattern> pattern) {
        auto action = GUI::Action::create(move(name), [this, pattern = pattern.ptr()](const GUI::Action&) {
            on_pattern_selection(pattern);
        });
        pattern->set_action(action);
        m_patterns.append(move(pattern));
    };

```

They still need to be moved here even if they are moved in the add_pattern lambda function?

Reply:
this is about the vectors that are passed to the Pattern constructor by value, and not the created objects - so by the time we do `m_pattern = move(pattern);`, a copy has already been created needlessly, unless the compiler can somehow figure this out on its own or I'm mistaken.

Reply:
Got you, that makes sense. Added a move to all of the vectors then.